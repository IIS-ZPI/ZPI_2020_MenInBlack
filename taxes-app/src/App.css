.App {
  text-align: center;
}

.App-header {
  font-size: 80px;
}
