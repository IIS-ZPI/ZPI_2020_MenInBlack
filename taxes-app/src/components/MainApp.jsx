import React, { Component } from 'react';
import FormComponent from './FormComponent';
import TableComponent from './TableComponent'
import '../css/form.css'

const statesApi = "http://localhost:4000/states"

class MainApp extends Component {
    constructor(props) {
        super();
        this.state = {
            displayTable: false,
            category: "groceries",
            amount: "0.00",
            data: []
        }
    }

    displayTable = () => {
        this.constructTableData();

        this.setState({
            displayTable: true
        })
    }

    hideTable = () => {
        this.setState({
            displayTable: false
        })
    }

    countNettoValue = (tax, desiredPrice) => {
        let divider = tax + 1;
        return this.formatPrice(desiredPrice / divider);
    }

    formatPrice = (number) => {
        return Number(number).toFixed(2);
    }

    constructTableData = () => {
        let iterator = 0;
        let resultsArray = [];
        let tempRecord;
        fetch(statesApi)
            .then(res => res.json())
            .then((
                result => {
                    for (var k in result) {
                        //get the tax in given state
                        let tax = +result[k][this.state.category] + +result[k].base;
                        //count netto prive
                        let nettoPrice = this.countNettoValue(tax, this.state.amount);
                        tempRecord = { id: iterator++, state: k, tax: tax + '%', netto: nettoPrice, margin: this.formatPrice(this.state.amount - nettoPrice) }
                        resultsArray.push(tempRecord);
                    };
                    this.setState({
                        data: resultsArray
                    })
                }
            ))
    }

    categoryChange = (category) => {
        this.setState({
            category
        })
    }

    amountChange = (amount) => {
        this.setState({
            amount
        })
    }

    render() {
        return (
            <div id="mainComponent">
                <FormComponent
                    hideTable={this.hideTable}
                    submitPress={this.displayTable}
                    categoryChange={this.categoryChange}
                    amountChange={this.amountChange}
                >
                </FormComponent>
                <br />
                {this.state.displayTable &&
                    <TableComponent
                        tableData={this.state.data}
                    >
                    </TableComponent>
                }
            </div>
        );
    }
}

export default MainApp;