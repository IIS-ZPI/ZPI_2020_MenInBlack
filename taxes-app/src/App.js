import React from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import MainApp from './components/MainApp';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        Welcome to taxes app!
      </header>
      <br />
      <MainApp />
    </div >
  );
}

export default App;
