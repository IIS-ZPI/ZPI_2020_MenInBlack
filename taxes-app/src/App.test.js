import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import MainApp from './components/MainApp';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


test('renders taxes app text', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Welcome to taxes app!/i);
  expect(linkElement).toBeInTheDocument();
});

describe('MainApp', () => {
  it('should render correctly', () => {
    const component = shallow(<MainApp />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<MainApp />);

    expect(component).toMatchSnapshot();
  });

  it("should count netto price properly", () => {
    //given
    const wrapper = shallow(<MainApp />);
    let desiredPrice = 10.00;
    let tax = 0.04;
    //when
    let nettoPrice = wrapper.instance().countNettoValue(tax, desiredPrice)
    //then
    expect(nettoPrice).toBe("9.62");
  });
});
