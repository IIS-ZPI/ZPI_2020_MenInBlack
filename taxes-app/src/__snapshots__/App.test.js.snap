// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp should render correctly 1`] = `ShallowWrapper {}`;

exports[`MainApp should render correctly with no props 1`] = `ShallowWrapper {}`;
